### PR TITLE
mas: init at 1.8.1

### PR DESCRIPTION
**Pull request by zachcoyle**
###### Motivation for this change
mas is a command line interface to the Mac App store, and having this package available will allow automation of interactions with the App Store

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/darwin/mas/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{ lib
+, stdenv
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/mas/default.nix:2]:
```suggestion
{ stdenv
```

File: pkgs/os-specific/darwin/mas/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  };
+
+  buildInputs = [ libarchive p7zip ];
+
+  unpackPhase = ''
+    7z x $src
+    bsdtar -xf Payload~
+  '';
+
+  buildPhase = "true";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/mas/default.nix:24]:
```suggestion
  doBuild = false;
```

File: pkgs/os-specific/darwin/mas/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Mac App Store command line interface";
+    homepage = "https://github.com/mas-cli/mas";
+    license = licenses.mit;
+    maintainers = with maintainers; [ zachcoyle ];
+    platforms = platforms.darwin;
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/mas/default.nix:44]:
```suggestion
```

File: pkgs/os-specific/darwin/mas/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    cp -r ./bin $out
+    cp -r ./Frameworks $out
+  '';
+
+  postFixup = ''
+    install_name_tool -change @rpath/MasKit.framework/Versions/A/MasKit $out/Frameworks/MasKit.framework/Versions/A/MasKit $out/bin/mas
+    install_name_tool -change @rpath/Commandant.framework/Commandant $out/Frameworks/MasKit.framework/Versions/A/Frameworks/Commandant.framework/Versions/A/Commandant $out/bin/mas
+  '';
+
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/mas/default.nix:37]:
```suggestion
  meta = with stdenv.lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please put this under os-specific.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99608` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mas</li>
  </ul>
</details>

**Comment by zachcoyle** [comment]
@SuperSandro2000 Thanks for the feedback! Somehow I didn't see this until now

File: pkgs/os-specific/darwin/mas/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+stdenv.mkDerivation rec {
+  pname = "mas";
+  version = "1.8.1";
+
+  src = fetchurl {
+    url = "https://github.com/mas-cli/mas/releases/download/v${version}/mas.pkg";
+    sha256 = "W/wgg+ETeJPoZ7MoVGH2uJzQiZMLIy3n1JYKUloc3ZU=";
+  };
+
+  buildInputs = [ libarchive p7zip ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/mas/default.nix:16]:
```suggestion
  nativeBuildInputs = [ libarchive p7zip ];
```

File: pkgs/os-specific/darwin/mas/default.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    cp -r ./bin $out
+    cp -r ./Frameworks $out
+  '';
+
+  postFixup = ''
+    install_name_tool -change @rpath/MasKit.framework/Versions/A/MasKit $out/Frameworks/MasKit.framework/Versions/A/MasKit $out/bin/mas
+    install_name_tool -change @rpath/Commandant.framework/Commandant $out/Frameworks/MasKit.framework/Versions/A/Frameworks/Commandant.framework/Versions/A/Commandant $out/bin/mas
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/mas/default.nix:36]:
```suggestion
  meta = with lib; {
```

File: pkgs/os-specific/darwin/mas/default.nix [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{ stdenv
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/mas/default.nix:1]:
```suggestion
{ lib
, stdenv
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 99608` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>mas</li>
</ul>
</details>

**Pull request merged**